test/misc/output-options-pattern-function.js

```diff
@@ -0,0 +1,19 @@
+const assert = require('assert');
+const rollup = require('../../dist/rollup');
+const { loader } = require('../utils.js');
+
+describe('options.output.pattern', () => {
```

Comment:
While it is technically possible, you should only use `describe/it` based tests for Rollup if none of the other formats fit as they are harder to maintain and contain a lot of noise. Also as a side note, this test is not run by the system because the file will not be picked up automatically, as you can see by looking at the coverage.

For the current feature, I would recommend adding a test to the `chunking-form` folder. For the particular case, I would suggest to duplicate the `chunking-form/emit-asset/asset-file-names` test as it contains both an asset and a chunk. Maybe extend it with a dynamic import to have a dynamic chunk that uses the `chunkFileNames` pattern as well.

Reply:
@lukastaegert I will copy the filenames-patterns test as our function will return a pattern. i will call it filenames-function-pattern do you know a way to run a single test ? I did the mocha single test as this was easy run-able independent at last for me

Reply:
Just add `solo: true` to the `_config.js` file